src/Safebox.sol

```diff
@@ -0,0 +1,197 @@
+     * @param amount The amount of tokens.
+     */
+    function withdraw(uint256 amount) external override {
+        require(wards[msg.sender] == 1 || vat.live() == 0, "Safebox/not-ward");
+
+        token.transfer(recipient, amount);
+        emit Withdraw(recipient, amount);
+    }
+
+    /**
+     * @notice Deposits tokens into this contract.
+     * @param amount The amount of tokens.
+     */
+    function deposit(uint256 amount) external override {
+        token.transferFrom(msg.sender, address(this), amount);
+        emit Deposit(msg.sender, amount);
+    }
+
+    /*//////////////////////////////////
+            MakerDAO Interfaces
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` owner access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external override auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` owner access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external override auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"recipient"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external override auth {
+        require(vat.live() == 1, "Safebox/vat-not-live");
+
+        if (what == "recipient") {
+            require(data != address(0), "Safebox/invalid-recipient");
+            pendingRecipient = data;
+            emit File(what, data);
+        } else {
+            revert("Safebox/file-unrecognized-param");
+        }
+    }
+
+    /*//////////////////////////////////
+            Custodian Interfaces
+    //////////////////////////////////*/
+
+    /**
+     * @notice Check if an address has `owner` access on this contract.
+     * @param usr The user address.
+     * @return Whether `usr` is a ward or not.
+     */
+    function isOwner(address usr) external view override returns (bool) {
+        return wards[usr] == 1;
+    }
+
+    /**
+     * @notice Check if an address has `custodian` access on this contract.
+     * @param usr The user address.
+     * @return Whether `usr` is a ward or not.
+     */
+    function isCustodian(address usr) external view override returns (bool) {
+        return custodians[usr] == 1;
+    }
+
+    /**
+     * @notice Adds a new custodian to this contract.
+     * @param usr The user address.
+     */
+    function addCustodian(address usr) external override onlyCustodian {
+        custodians[usr] = 1;
+        emit AddCustodian(usr);
+    }
+
+    /**
+     * @notice Removes a custodian from this contract.
+     * @param usr The user address.
+     */
+    function removeCustodian(address usr) external override onlyCustodian {
```

Comment:
Note that as with `wards` behavior, also here if there is more than one custodian this will allow any custodian to remove any other.
So need to make sure this is wanted.